mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -303,6 +322,29 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group setParam */
+  @Since("2.1.0")
+  def setInitialModel(value: KMeansModel): this.type = set(initialModel, value)
+
+  /** @group setParam */
+  @Since("2.1.0")
+  def setInitialModel(value: Model[_]): this.type = {
+    value match {
+      case m: KMeansModel => setInitialModel(m)
+      case other =>
+        logWarning(s"KMeansModel required but ${other.getClass.getSimpleName} found.")
```

Comment:
I prefer an error here. Other parameters throw errors when invalid values are set, so I don't see any reason to deviate from that behavior here.